manager/filters.py

```diff
@@ -512,6 +512,48 @@ def _filter_system_cve_by_remediation(query, args, _kwargs):
     return query
 
 
+def _filter_system_by_ansible(query, args, _kwargs):
+    """
+    Filters list of systems based if system is managed by Ansible Automation Platform.
+    This information is stored inside cyndi view.
+    Query needs to be joined with cyndi inventory.hosts view!
+
+    Args:
+        query (object): Query object to apply filters to
+        args (dict): Query arguemnts
+
+    Returns:
+        object: Modified query with AAP filter applied
+    """
+    if 'ansible' in args and args['ansible'] is not None:
+        expr = InventoryHosts.system_profile.contains({'ansible'})
+        if not args['ansible']:
+            expr |= ~(InventoryHosts.system_profile.contains({'ansible'}))
+        query = query.where(expr)
+    return query
+
+
+def _filter_system_by_mssql(query, args, _kwargs):
+    """
+    Filters list of systems based if system is managed by MSSQL.
+    This information is stored inside cyndi view.
+    Query needs to be joined with cyndi inventory.hosts view!
+
+    Args:
+        query (object): Query object to apply filters to
+        args (dict): Query arguemnts
+
+    Returns:
+        object: Modified query with MSSQL filter applied
+    """
+    if 'mssql' in args and args['mssql'] is not None:
+        expr = InventoryHosts.system_profile.contains({'version'})
```

Comment:
Detecting 'version' key is too general. I can hit also different sub-keys in system_profile not related to mssql. For example rhsm version field https://github.com/RedHatInsights/inventory-schemas/blob/master/schemas/system_profile/v1.yaml#L471